persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/StargateQueryHandler.java

```diff
@@ -64,15 +87,188 @@ public ResultMessage process(
               statement.getBoundTerms(), options.getValues().size()));
     }
 
-    if (!queryState.getClientState().isInternal)
+    if (!queryState.getClientState().isInternal) {
       QueryProcessor.metrics.regularStatementsExecuted.inc();
+    }
 
     ResultMessage result =
         maybeIntercept(statement, queryState, options, customPayload, queryStartNanoTime);
-    return result == null
-        ? QueryProcessor.instance.processStatement(
-            statement, queryState, options, queryStartNanoTime)
-        : result;
+
+    if (result != null) {
+      return result;
+    }
+
+    if (customPayload != null && customPayload.containsKey("token")) {
+      authorizeByToken(customPayload.get("token"), statement);
```

Comment:
Could re-parsing the token on every message be expensive and affect performance?

Reply:
Even though, aub-classing `AuthenticatedUser` might cause some extra work in the DSE Persistence, but it should allow us to cache pre-parsed auth data (minimizing per-message work)... WDYT?